From 72c98ae6961f7c8a8e5e00bddd6fb08ca850b479 Mon Sep 17 00:00:00 2001
Message-Id: <72c98ae6961f7c8a8e5e00bddd6fb08ca850b479.1350493760.git.minovotn@redhat.com>
In-Reply-To: <c93188b6c6973932d2adaa52e6a4920db13b4e62.1350493760.git.minovotn@redhat.com>
References: <c93188b6c6973932d2adaa52e6a4920db13b4e62.1350493760.git.minovotn@redhat.com>
From: Jeffrey Cody <jcody@redhat.com>
Date: Wed, 17 Oct 2012 05:59:39 +0200
Subject: [PATCH 26/35] block: remove keep_read_only flag from
 BlockDriverState struct

RH-Author: Jeffrey Cody <jcody@redhat.com>
Message-id: <fb1faca3f27013bf871842d97828361c8812b169.1350447475.git.jcody@redhat.com>
Patchwork-id: 43283
O-Subject: [RHEL6.4 qemu-kvm PATCH v4 26/35] block: remove keep_read_only flag from BlockDriverState struct
Bugzilla: 767233
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Eric Blake <eblake@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

The keep_read_only flag is no longer used, in favor of the bdrv
flag BDRV_O_ALLOW_RDWR.

Signed-off-by: Jeff Cody <jcody@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit dc1c13d96912731d4c7c7e13d31c93b8735f1203)
---
 block.c     | 2 --
 block_int.h | 1 -
 2 files changed, 3 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 block.c     | 2 --
 block_int.h | 1 -
 2 files changed, 3 deletions(-)

diff --git a/block.c b/block.c
index 17d3886..026e528 100644
--- a/block.c
+++ b/block.c
@@ -651,8 +651,6 @@ int bdrv_open(BlockDriverState *bs, const char *filename, int flags,
         flags |= BDRV_O_ALLOW_RDWR;
     }
 
-    bs->keep_read_only = !(flags & BDRV_O_ALLOW_RDWR);
-
     /* Open the image */
     ret = bdrv_open_common(bs, filename, flags, drv);
     if (ret < 0) {
diff --git a/block_int.h b/block_int.h
index 883fd5f..739c5fc 100644
--- a/block_int.h
+++ b/block_int.h
@@ -228,7 +228,6 @@ struct BlockDriverState {
     int64_t total_sectors; /* if we are reading a disk image, give its
                               size in sectors */
     int read_only; /* if true, the media is read only */
-    int keep_read_only; /* if true, the media was requested to stay read only */
     int open_flags; /* flags used to open the file, re-used for re-open */
     int encrypted; /* if true, the media is encrypted */
     int valid_key; /* if true, a valid encryption key has been set */
-- 
1.7.11.7

